src/app/activities/activity-list/activity-forecast-list.component.ts

```diff
@@ -15,24 +16,57 @@ export class ActivityForecastListComponent implements OnInit {
   today: Date = new Date();
   msInaDay: number = 24 * 3600 * 1000;
 
-  constructor(private route: ActivatedRoute, private alerts: AlertsService) { }
+  private BASE_HUB_URL: string;
+
+  constructor(private route: ActivatedRoute, appConfig: AppConfigService, private alerts: AlertsService) {
+    this.BASE_HUB_URL = appConfig.config.HUB_URL;
+  }
 
   ngOnInit() {
     this.route.data.subscribe(data => {
-      if(typeof data['activities'] === 'undefined' || data['activities'] === null) {
+      if (typeof data['activities'] === 'undefined' || data['activities'] === null) {
         this.alerts.showError('An error occurred while retrieving activities');
         return;
-      };
+      }
       let todayDow = this.today.getDay();
       if (todayDow === 6) { todayDow = 0; } // group Sunday with Saturday
       data['activities'].forEach(v => {
+        this.processHqComments(v);
+
         v.startDateTime = new Date(v.startDateTime);
         let dow = v.startDateTime.getDay();
         if (dow === 6) { dow = 0; } // group Sunday with Saturday
         this.activitiesPerDays[dow >= todayDow ? dow - todayDow : dow + 6 - todayDow].push(v);
       });
     });
   }
+
+  processHqComments(activity: Activity) {
```

Comment:
can you add some tests for this function? it's a bit of a beast so it would be good to know it does what we think it does

Reply:
I also wonder if theres something built in that can be used for this? Don't think we're the first people to be parsing markdown. Is there anything special about these comments that would prevent that?